pymc3/distributions/multivariate.py

```diff
@@ -954,42 +960,101 @@ def _lkj_normalizing_constant(eta, n):
     return result
 
 
+class _LKJCholeskyCovRV(RandomVariable):
+    name = "_lkjcholeskycov"
+    ndim_supp = 0
+    ndims_params = [0, 0, 0]
+    dtype = "floatX"
+    _print_name = ("_LKJCholeskyCov", "\\operatorname{_LKJCholeskyCov}")
+
+    @classmethod
+    def rng_fn(cls, rng, eta, n, sd_dist, size=None):
+        n = int(n)
+        eta = int(eta)
+        dist_shape = ((n * (n + 1)) // 2,)
+
+        # We will always provide random with an integer size and then reshape
+        # the output to get the correct size
+        if size is not None:
+            orig_size = size
+            size = np.prod(size)
+        else:
+            size = 1
+            orig_size = None
+
+        P = np.eye(n) * np.ones((size,) + (n, n))
+        # original implementation in R see:
+        # https://github.com/rmcelreath/rethinking/blob/master/R/distributions.r
+        beta = eta - 1.0 + n / 2.0
+        r12 = 2.0 * stats.beta.rvs(a=beta, b=beta, size=(size,)) - 1.0
+        P[..., 0, 1] = r12
+        P[..., 1, 1] = np.sqrt(1.0 - r12 ** 2)
+        for mp1 in range(2, n):
+            beta -= 0.5
+            y = stats.beta.rvs(a=mp1 / 2.0, b=beta, size=(size,))
+            z = stats.norm.rvs(loc=0, scale=1, size=(size,) + (mp1,))
+            z = z / np.sqrt(np.einsum("ij,ij->j", z, z))
+            P[..., 0:mp1, mp1] = np.sqrt(y[..., np.newaxis]) * z
+            P[..., mp1, mp1] = np.sqrt(1.0 - y)
+        C = np.einsum("...ji,...jk->...ik", P, P)
+
+        D = sd_dist
```

Comment:
Previously this drew from `.random` of `sd_dist`. But we can make sure the output shape of the `rv_out` in `sd_dist` is `(size, n)`. Which can be directly assigned over here.